pymc/aesaraf.py

```diff
@@ -537,23 +537,64 @@ def join_nonshared_inputs(
     point: Dict[str, np.ndarray],
     xs: List[TensorVariable],
     vars: List[TensorVariable],
-    shared,
+    shared: Dict[TensorVariable, TensorSharedVariable],
```

Comment:
In the first pure aesara example you provided, the shared dict is TensorVariable to `aesara.tensor.sharedvar.ScalarSharedVariable`. Which works fine. 
Would it be accurate to include `Union[TensorSharedVariable, ScalarSharedVariable]` as this type hint?